debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -360,7 +361,9 @@ private void initializeRedoLogsForMining(OracleConnection connection, boolean po
             if (!isContinuousMining) {
                 currentLogFiles = setLogFilesForMining(connection, startScn, archiveLogRetention, archiveLogOnlyMode,
                         archiveDestinationName, logFileQueryMaxRetries, initialDelay, maxDelay);
-                currentRedoLogSequences = getCurrentLogFileSequences(currentLogFiles);
+                if (!archiveLogOnlyMode) {
```

Comment:
To add to what Voljtech said, the log switch calculation is essential even for archive log mode because we need to know when a new archive log has been generated. We only know this when we detect log switches, so this step is critical regardless of the mode used.